video/out/wayland_common.c

```diff
@@ -1703,21 +1726,35 @@ static void set_content_type(struct vo_wayland_state *wl)
 #endif
 }
 
+static void set_cursor_shape(struct vo_wayland_state *wl)
+{
+    if (!wl->cursor_shape_device)
```

Comment:
The only place where this function gets called from guarantees the existence of this object. The check should be removed.

Reply:
Thanks for looking!